doc/article/en-US/testing-components.md

```diff
@@ -164,7 +226,97 @@ Now, let's assert that the element actually gets the background color it is boun
 
 As you can see, everything follows the same pattern we had for our custom element test. One exception is that we take advantage of the `element` property which gets provided by the `ComponentTester` instance. The `element` property is the actual HTML element that gets rendered. This can also be used when testing custom elements.
 
-## [Helpful Properties and Functions](aurelia-doc://section/5/version/1.0.0)
+## [Using a Real Parent View-model](aurelia-doc://section/6/version/1.0.0)
+
+If you want to test using a custom element inside of a real parent view-model this can be done just as easily.  This can be really helpful when needing to test the state of a parent that is affected by the child custom element or attribute - 
+
+<code-listing heading="A Custom Attribute Test with Real Parent View-model">
+  <source-code lang="JavaScript">
+    import {StageComponent} from 'aurelia-testing';
+    import {bootstrap} from 'aurelia-bootstrapper';
+    import {MyComponent} from 'src/my-component';
+
+    describe('MyAttribute', () => {
+      let component;
+      let viewModel;
+
+      beforeEach(() => {
+        viewModel = new MyComponent();
+        component = StageComponent
+            .withResources('src/my-attribute')
+            .inView('<div my-attribute.bind="color">Bob</div>')
+            .boundTo(viewModel);
+      });
+      //...
+    });
+  </source-code>
+</code-listing>
+
+Using this you can also use the `ref` custom attribute to get access to things and check their state in the view-model.
+
+Or if your view-model has dependencies to load through DI - 
+
+<code-listing heading="A Custom Attribute Test with Real Parent View-model with DI dependencies">
+  <source-code lang="JavaScript">
+    import {StageComponent} from 'aurelia-testing';
+    import {bootstrap} from 'aurelia-bootstrapper';
+    import {MyComponent} from 'src/my-component';
+    import {Container} from 'aurelia-dependency-injection';
+    import {MyService} from 'src/my-service';
+
+    describe('MyAttribute', () => {
+      let component;
+      let container;
+      let viewModel;
+      let myService;
+
+      beforeEach(() => {
+        container = new Container().makeGlobal();
```

Comment:
I'd like to avoid even showing people the `makeGlobal()` API since we don't want that to get abused. There should be a better way to handle this. @martingust @jdanyow isn't there a way to get access to the container that is going to be used so that you can set it up? If not, I think we should add a hook for that.